# Import Library

Menginstal library dan modul yang akan digunakan

In [12]:
!pip install -q split-folders

Menghubungkan Google Drive dengan Google Colab

In [13]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Import Data from Drive

Mengakses dataset dari Google Drive

In [14]:
dataset = '/content/drive/MyDrive/Breast Cancer Dataset'

Melakukan pembagian data dengan ratio 80% (train) dan 20% (test)

In [15]:
#impor package split-folders
import splitfolders

#Membagi data
splitfolders.ratio(dataset,
                   output='/content/drive/MyDrive/Breast Cancer Dataset/data',
                   seed=1337,
                   ratio=(.8, .2),
                   group_prefix=None, move=False)

Copying files: 481 files [00:16, 28.57 files/s]


# Split Data

In [16]:
import os

base_dir = '/content/drive/MyDrive/Breast Cancer Dataset/data'

train_dir = os.path.join(base_dir, 'train')
os.chdir(base_dir)
os.rename("val","test")
test_dir = os.path.join(base_dir, 'test')

train_benign = '/content/drive/MyDrive/Breast Cancer Dataset/data/train/benign'
train_malignant = '/content/drive/MyDrive/Breast Cancer Dataset/data/train/malignant'
train_normal = '/content/drive/MyDrive/Breast Cancer Dataset/data/train/normal'

test_benign = '/content/drive/MyDrive/Breast Cancer Dataset/data/test/benign'
test_malignant = '/content/drive/MyDrive/Breast Cancer Dataset/data/test/malignant'
test_normal = '/content/drive/MyDrive/Breast Cancer Dataset/data/test/normal'

print('total citra train:', len(os.listdir(train_benign))+
                            len(os.listdir(train_malignant))+
                            len(os.listdir(train_normal))
                            )
print('total citra test:', len(os.listdir(test_benign))+
                           len(os.listdir(test_malignant))+
                           len(os.listdir(test_normal)))

total citra train: 384
total citra test: 97


Hasil pembagian data citra yang digunakan untuk train sebanyak 384 citra dan citra yang digunakan untuk test sebanyak 97 citra.